week-12/homework/index.js

```diff
@@ -0,0 +1,131 @@
+let input = document.querySelector('.input');
+let addBtn = document.querySelector('.add-btn');
+let itemContainer = document.querySelector('.item-container');
+let deleteBtns = document.querySelectorAll('.delete-btn');
+let editBtns = document.querySelectorAll('.edit-btn');
+let currentTaskNum = 0;
+
+//Creates tasklist, takes taskNumber and input value
+let processTaskList = function (taskNum, inputValue) {
+    let newInput = document.createElement("input");
+    let deleteBtn = document.createElement("button");
+    let editBtn = document.createElement("button");
+    let saveBtn = document.createElement("button");
+    let newDiv = document.createElement("div");
+
+    deleteBtn.innerHTML = "X";
+    deleteBtn.setAttribute('delete-item-number', taskNum);
+    deleteBtn.classList.add('delete-btn');
+    deleteBtn.classList.add('btn');
+    deleteBtn.classList.add('btn-danger');
+    editBtn.innerHTML = "Edit";
+    editBtn.setAttribute('edit-item-number', taskNum);
+    editBtn.classList.add('edit-btn');
+    editBtn.classList.add('btn');
+    editBtn.classList.add('btn-primary');
+    saveBtn.innerHTML = "Save";
+    saveBtn.setAttribute('save-item-number', taskNum);
+    saveBtn.classList.add('save-btn');
+    saveBtn.classList.add('d-none');
+    saveBtn.classList.add('btn');
+    saveBtn.classList.add('btn-secondary');
+    newInput.type = "text";
+    newInput.setAttribute('disabled', 'true');
+    newInput.setAttribute('input-item-number', taskNum);
+    newInput.classList.add('input-item');
+    newInput.classList.add('form-control');
+    newInput.value = inputValue;
+    newDiv.classList.add("item");
+    newDiv.classList.add("d-flex");
+    newDiv.classList.add("pt-1")
+    newDiv.setAttribute('item-number', taskNum);
+
+    newDiv.appendChild(newInput);
+    newDiv.appendChild(deleteBtn);
+    newDiv.appendChild(editBtn);
+    newDiv.appendChild(saveBtn);
+
+    itemContainer.appendChild(newDiv);
+
+
+    input.value = '';
+
+    deleteBtns = document.querySelectorAll('.delete-btn');
+    editBtns = document.querySelectorAll('.edit-btn');
+
+    for (let i = 0; i < deleteBtns.length; i++) {
```

Comment:
There is a much better way so you do not need to loop and at the same time ensure we attach the right event only once.

I list my suggestion below and you need to think about how you will execute it
- 1. When this function is execute, create an empty `div`
- 2. Use the approach `innerHTML` above to set HTML content you need to that `div`. PS. if you follow this way you do not need any data-attribute set to input or button or anything
- 3. Append the `div` above to `itemContainer`
- 4. Start selecting buttons and input inside that `div` using `newDiv.querySelector`
- 5. Step 4 allow us to select just the buttons and input inside that `newDiv` so you just need to call `deleteBtn.addEventListener('click', processDeleteBtn)` ... 
- Work out the remaining logic yourself